### ISSUE TITLE
years of species "discovery"

### ISSUE THREAD

**Issue by wcornwell** [tmesaglio/Australian-plant-photos#3]
![image](https://user-images.githubusercontent.com/5534161/155442848-9748627f-b6f3-4606-a1dd-68b9d4909f17.png)

**Comment by wcornwell** [comment]
![image](https://user-images.githubusercontent.com/5534161/155444747-20ee90e4-a129-40ce-967b-3a2615c436e3.png)

**Comment by wcornwell** [comment]
weeds versus native
![image](https://user-images.githubusercontent.com/5534161/155445191-fd85512d-6132-4994-a575-bb2435f601af.png)

**Comment by wcornwell** [comment]
the "primary reference" is mostly to floras and mostly for phrase names and hybrids

**Comment by tmesaglio** [comment]
ah makes sense

**Comment by tmesaglio** [comment]
geographic breakdown would be interesting. Presumably the big uptick post 70s is more expeditions to remote NT, northern QLD, etc.?

**Comment by wcornwell** [comment]
![image](https://user-images.githubusercontent.com/5534161/155446127-088c5f72-08a7-44c0-b5bf-a4c96f8e524b.png)

**Comment by wcornwell** [comment]
a bit bigger 90's-2000 spike, but still there for NSW

**Comment by tmesaglio** [comment]
really surprising to me for NSW

**Comment by wcornwell** [comment]
and Tas: 
![image](https://user-images.githubusercontent.com/5534161/155446358-1479906b-b365-4075-aa26-1040953f2741.png)

**Comment by tmesaglio** [comment]
perhaps a cool little side angle; how many of the species described in Brown 1810 have been photographed?! Do we think 100%?

**Comment by tmesaglio** [comment]
or perhaps a couple of them have gone extinct

**Comment by wcornwell** [comment]
Sloppy code here, for future reference
```{r}
library(tidyverse)

a<-read_csv("data/APNI-names-2021-06-08-1039.csv")

a %>%
  filter(taxonomicStatus=="accepted" & taxonRank=="Species") %>%
  filter(namePublishedInYear>0) ->z

sort(table(z$namePublishedInYear),decreasing = T)[1:10]

temp<-lapply(apc,FUN=function(x)grepl("native",x))

apc$native<-apply(bind_cols(temp),1,sum)>0

z %>% left_join(apc,by=c("scientificName"="species"))  %>% filter(Tas=="native") %>%
  ggplot(aes(x=namePublishedInYear,fill=Tas))+geom_histogram(bins=80)+theme_bw() + facet_grid(native~.,scales="free")

z %>% left_join(apc,by=c("scientificName"="species")) %>% filter(native==TRUE) ->zz
```

**Comment by wcornwell** [comment]
model for probability of a species have a photograph in iNat:

![image](https://user-images.githubusercontent.com/5534161/159583781-f9a52f06-776f-45d0-a6b5-9cb9c29242d7.png)

**Comment by wcornwell** [comment]
less wiggly model:
![image](https://user-images.githubusercontent.com/5534161/159584395-224889a3-9394-4730-8872-6fc563109cbc.png)

**Comment by tmesaglio** [comment]
very cool figures. Just to double check, you used the two files 'final_apc_matrix' and 'iM5' right?

**Comment by wcornwell** [comment]
opps i used iM4.  should i use iM5?

**Comment by tmesaglio** [comment]
yep, iM5 is the updated one I compiled last night after removing 300 or so non-natives that slipped in

**Comment by tmesaglio** [comment]
I also need to very slightly tinker with final_apc_matrix today, but that will only change 10-15 species. iM5 is ready to go though

**Comment by wcornwell** [comment]
ah OK. I'm removing not natives from `final_apc_matrix` so should come out the same.

**Comment by tmesaglio** [comment]
nah it won't be the same, there are no non-natives in final_apc_matrix; I have that file now as only containing natives.

**Comment by tmesaglio** [comment]
final_apc_matrix = master file of all native Australian species, with non-native, hybrids, phrase names removed
iM5 = master file of photographed native species on iNat, with adjusted names to match those in the apc matrix

**Comment by wcornwell** [comment]
ok will re-run.  in the meantime 166 species from Brown 1810 still haven't been photographed (out of 926 still valid names).

**Comment by tmesaglio** [comment]
wow, that seems like a crazy high amount. I would assume a lot of grasses and/or assumed extinct things in there?

**Comment by wcornwell** [comment]
a lot of small range endemics from WA:
https://bie.ala.org.au/species/https://id.biodiversity.org.au/node/apni/7830778

**Comment by tmesaglio** [comment]
interesting those have no iNat pics given Stirling Range NP is probably one of the better covered areas in WA on iNat. Both have pics online outside iNat though luckily

**Comment by wcornwell** [comment]
pretty similar using iM5:
![image](https://user-images.githubusercontent.com/5534161/159593729-a5f75a5d-c414-43f2-ad64-a10846114b07.png)

**Comment by wcornwell** [comment]
```{r}
#ploting date of species naming

library(tidyverse)

all<-read_csv("data/APNI-names-2022-03-23-0843.csv")

all %>%
  filter(nameType=="scientific") %>%
  filter(taxonomicStatus=="accepted") %>%
  filter(taxonRank=="Species") %>%
  filter(nameInstanceType=="tax. nov.") %>%
  select(canonicalName,namePublishedInYear) -> only_new_sp_names

finalmatrix <- read_csv("data/final_apc_matrix.csv")
iM5 <- read_csv("data/iM5.csv")

get_native_list<-function(finalmatrix=finalmatrix){
  counter<-finalmatrix$NT=="native" 
  for(i in 4:20){
    counter<-counter+as.vector(finalmatrix[,i]=="native")
  }
  return(data.frame(cName=finalmatrix$canonicalName[counter>0]))
}

native_list <- get_native_list(finalmatrix)
native_list$photo_yes_no <- native_list$cName %in% iM5$APC_name

native_list %>%
  left_join(only_new_sp_names, by=c("cName"="canonicalName")) -> graphing_df

graphing_df %>%
  ggplot(aes(x=namePublishedInYear,fill=photo_yes_no))+geom_histogram()+theme_bw()


graphing_df$photo_binary<-as.numeric(graphing_df$photo_yes_no)
graphing_df %>%
  ggplot(aes(x=namePublishedInYear,y=photo_binary))+geom_smooth()+theme_bw()

graphing_df %>%
  ggplot(aes(x=namePublishedInYear,y=photo_binary))+geom_jitter(alpha=.01,height=0.03) +geom_smooth(method="glm", method.args = list(family=binomial))+theme_bw()
```

**Comment by wcornwell** [comment]
to get whole list for 1810:
```
described_in_1810_and_still_not_photographed <-
  filter(graphing_df,
         namePublishedInYear == "1810",
         photo_yes_no == "FALSE")
```

**Comment by tmesaglio** [comment]
let me do the very minor cleaning now of final_apc_matrix (won't impact the shape of the figures above), and then the iNat part will be done. Then I can think about how to get names from flickr, APII, etc

**Comment by wcornwell** [comment]
> Both have pics online outside iNat though luckily

let's start a new issue on that

**Comment by eflowerproject** [comment]
This is all fascinating! I'm quite surprised by the millenial gap. It is quite possible that it is a matter of mostly rare species that are less likely to end up in iNat, but would still be likely to have been photographed by the authors with the photos sitting on a computer and/or a local website and/or as a plate in the original description. The question of the Brown photos is a great one too and deifnitely an interesting side story for the paper!

**Comment by wcornwell** [comment]
Apparently Brown's first stop in Aus was at King George Sound where Albany, WA is today--so I guess it makes sense that he collected a bunch of Stirling Range endemics...
